### ISSUE TITLE
10$: don't error on class with same name as aliased empty namespace

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#1122]
quick-lint-js correctly detects empty TypeScript namespaces and does not report a diagnostic if the namespace's name conflicts with another variable:
```typescript
namespace C {}  // empty
class C {}  // no conflict
```
However, quick-lint-js is not smart enough when namespace aliases are involved:
```typescript
namespace C {}  // empty
import D = C;
class D {}  // no conflict, but quick-lint-js diagnoses
```
